odoo/fields.py

```diff
@@ -2057,8 +2057,13 @@ def process(ids):
         elif isinstance(value, (list, tuple)):
             # value is a list/tuple of commands, dicts or record ids
             comodel = record.env[self.comodel_name]
-            # determine the value ids; by convention empty on new records
-            ids = OrderedSet(record[self.name].ids if record.id else ())
+            # determine the value ids
+            if record.id and record[self.name].ids:
+                ids = OrderedSet(record[self.name].ids)
+            elif type(record.id) in IdType and record[self.name]._ids:
+                ids = OrderedSet(record[self.name]._ids)
+            else:
+                ids = OrderedSet()
```

Comment:
Why not this?
```python
ids = OrderedSet(record[self.name]._ids)
```

Reply:
Ah, yes, that would be ok.

Reply:
It's ok in stable? If yes, then I will update this PR. If not, then I let to you to cheery-pick haha

Reply:
updated